source/plugins/go/src/ingestion_token_utils.go

```diff
@@ -450,7 +450,7 @@ func getAgentConfiguration(imdsAccessToken string) (configurationId string, chan
 
 	configurationId = agentConfiguration.Configurations[0].Configurationid
 	channelId = agentConfiguration.Configurations[0].Content.Channels[0].ID
-	if len(agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights) > 0 {
+	if !ContainerLogV2ConfigMap && len(agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights) > 0 {
 		ContainerLogSchemaV2 = agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights[0].Extensionsettings.DataCollectionSettings.EnableContainerLogV2
```

Comment:
you should add nil & empty check

Reply:
responded in this PR: https://github.com/microsoft/Docker-Provider/pull/999
Extensionsettings and DataCollectionSettings are embedded structs, not pointers to structs or slices/arrays. struct fields will be automatically initialized to their zero values
so if not set, the output of agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights[0].Extensionsettings will be { { "" "" nil false } }

if we need access the other field in dcr settings then we need to do the check on those feilds

Reply:
are you saying, if there is no extensionSettings or DataCollectionSettings in the DCR , this will  be return to false "agentConfiguration.Configurations[0].Content.Extensionconfigurations.Containerinsights[0].Extensionsettings.DataCollectionSettings.EnableContainerLogV2", is that right? Did you valid just enabling to MSI cluster without data collection settings?

Reply:
tested with s4 and s5 attached the backdoor yaml screenshot without dcr settings